nestedtensor/csrc/functions.cpp

```diff
@@ -95,8 +95,34 @@ Tensor NestedTensor_max_pool2d(
     IntArrayRef dilation,
     bool ceil_mode) {
   auto self_impl = get_nested_tensor_impl(self);
-  auto self_data = self_impl->_data;
-  auto structure = self_data.get_structure();
+  auto nt = self_impl->_data;
+  auto tensor_node = nt.get_structure();
+
+  bool all_sizes_same = true;
```

Comment:
I think this kind of pattern will pop up more frequently. Maybe we need a method like "is_tensor_shape"?